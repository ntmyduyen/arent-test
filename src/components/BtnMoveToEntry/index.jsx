import { Wrap } from "./index.styles"

const BtnMoveToEntry = ({ id, title, desc, url }) => {
    const _handleClick = () => {
        const entry = document.querySelector(`#${id}`)
        if(!entry) return;
        entry.scrollIntoView()
    }

    return (
        <Wrap onClick={_handleClick}>
            <img src={url} alt="entry" loading="lazy" />
            <div className="wrap-content">
                <div className="title">{title}</div>
                <div className="desc">{desc}</div>
            </div>
        </Wrap>
    )
}
export default BtnMoveToEntry