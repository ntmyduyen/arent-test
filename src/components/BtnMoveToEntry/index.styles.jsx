import styled from 'styled-components'

const Wrap = styled.div`
    width: 288px;
    height: 288px;
    background: var(--color-dark-600);
    border: 24px solid var(--color-primary-300);
    position: relative;
    cursor: pointer;

    img {
        position: absolute;
        width: 100%;
        height: 100%;
        object-fit: cover;
        z-index: 0;
        mix-blend-mode: luminosity;
    }

    .wrap-content {
        position: relative;
        width: 100%;
        height: 100%;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        background: rgba(0, 0, 0, 0.5);

        .title {
            font-family: var(--secondary-font);
            font-style: normal;
            font-weight: 400;
            font-size: 25px;
            line-height: 30px;
            letter-spacing: 0.125px;
            color: var(--color-primary-300);
        }
    
        .desc {
            margin-top: 10px;
            min-width: 160px;
            min-height: 24px;
            background: var(--color-primary-400);
            font-style: normal;
            font-weight: 300;
            font-size: 14px;
            line-height: 20px;
            text-align: center;
            color: var(--color-light);
        }
    }

`;

export {
    Wrap
}