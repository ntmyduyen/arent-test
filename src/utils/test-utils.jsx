/* eslint-disable import/export */
import setupStore from '@redux/index';
import { cleanup, render } from '@testing-library/react'
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { afterEach } from 'vitest'

afterEach(() => {
	cleanup()
})

function renderWithProviders(
	ui,
	{
		preloadedState = {},
		// Automatically create a store instance if no store was passed in
		store = setupStore(preloadedState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>
			<MemoryRouter>{children}</MemoryRouter>
		</Provider>
	}
	return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
export {
	renderWithProviders
}