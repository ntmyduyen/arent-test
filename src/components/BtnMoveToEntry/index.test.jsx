import { renderWithProviders } from "@utils/test-utils"
import BtnMoveToEntry from "."

describe("BtnMoveToEntry", () => {
    it('should be match with snapshot', () => {
        const { container } = renderWithProviders(<BtnMoveToEntry />)
        expect(container).toMatchSnapshot()
    })
})