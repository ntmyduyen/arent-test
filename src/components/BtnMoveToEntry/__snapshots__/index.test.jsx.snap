// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BtnMoveToEntry > should be match with snapshot 1`] = `
<div>
  <div
    class="sc-beqWaB jlGERk"
  >
    <img
      alt="entry"
      loading="lazy"
    />
    <div
      class="wrap-content"
    >
      <div
        class="title"
      />
      <div
        class="desc"
      />
    </div>
  </div>
</div>
`;
